File: test/shared.php
<?php

error_reporting(E_ALL | E_STRICT);

require_once(__DIR__ . '/../php/flatted.php');

$str = file_get_contents(__DIR__ . '/shared.json');
$json = Flatted::parse($str);

$arr = $json[0];
$obj = $json[1];

for ($i = 0; $i < 64; $i++) {
  $arr = $arr[0];
  $obj = $obj->obj;
}

assert(count($arr) == 2 && $arr[0] == 'arr' && $arr[1] == 1 && $obj->obj == 2);

assert(Flatted::stringify($json) == $str);

?>

File: test/shared.py
# ⚠️ RUN node test/shared.cjs FIRST
# then python test/shared.py

import os, sys

__dir__ = os.path.dirname(__file__)
sys.path.append(os.path.join(__dir__, '..', 'python'))

from flatted import parse, stringify

with open(os.path.join(__dir__, 'shared.json'), 'r') as f:
  str = f.read()

arr, obj = parse(str)

for i in range(64):
  arr = arr[0]
  obj = obj['obj']

assert len(arr) == 2 and arr[0] == 'arr' and arr[1] == 1 and obj['obj'] == 2

assert stringify(parse(str), separators=(',', ':')) == str
